### PR TITLE
CI: For macos, use 13

### PR DESCRIPTION
**Pull request by andy5995**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       run: ./configure && make && make test -k
 
   test-x86_64-osx:
-    runs-on: macos-11
```
Comment (andy5995) [inline .github/workflows/build.yml:35]: I noticed this job is stalled in the CI. Looks like 11 isn't even available anymore. At some point, we may want to add a matrix to test on more than one version, but this should be good enough for now (unless right now you'd prefer to "hard-code" a version).

https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

Comment (andy5995) [reply .github/workflows/build.yml:35]: I see 12 being used on the tinycc repo[mirror https://github.com/TinyCC/tinycc/blob/mob/.github/workflows/build.yml#L17](https://repo.or.cz/tinycc.git/blob/45cff8f03fe58a1ded52864bb6640998ea6775e5:/.github/workflows/build.yml) but that's deprecated according to the runner image docs, so I'm gonna switch it to 13 for now https://github.com/actions/runner-images

Comment (KaruroChori) [reply .github/workflows/build.yml:35]: Thank you for the information. Please, feel free to proceed as you think it is best :D!
I myself never used macos before, so I am totally unqualified to make decisions revolving around it.

Comment (andy5995) [reply .github/workflows/build.yml:35]: I don't use it either. I know just enough to use in tests ;)

**Pull request merged**